### PR TITLE
Offline mode

### PR DESCRIPTION
**Pull request by fredrikekre**

### CODE REVIEW

**Comment by KristofferC** [comment]
Can we merge this?

**Comment by fredrikekre** [comment]
I should add some tests, and also not sure if anything from the artifacts PR needs to be updated

**Comment by KristofferC** [comment]
Yeah, true.

**Comment by cossio** [comment]
This would be nice to have. Is there something holding this PR back?

**Comment by fredrikekre** [comment]
I'll revive it for 1.5.

**Comment by oschulz** [comment]
> I'll revive it for 1.5.

Lot's of thanks in advance, this will be very useful for us here!

**Comment by fredrikekre** [comment]
Rebased. This does not deal with cached clones, but I think it is good enough for now.

File: test/new.jl [hunk 1]
```diff
@@ -2375,4 +2375,23 @@ end
         Pkg.develop(Pkg.PackageSpec(path="B"))
     end
 end
+
+@testset "Offline mode" begin
+    isolate(loaded_depot=false) do
+        # cache this version
+        Pkg.add(PackageSpec(uuid=exuuid, version=v"0.5.1"))
+        @test Pkg.dependencies()[exuuid].version == v"0.5.1"
+        Pkg.offline()
```
Comment (KristofferC) [inline test/new.jl:2384]: Need to turn off offline mode after this test?

Comment (KristofferC) [reply test/new.jl:2384]: Also need to use `Pkg.PackageSpec` or `import it`.

**Comment by timholy** [comment]
This PR is a gift waiting to be unwrapped when we can all return to public transit. Thanks @fredrikekre! (And also @KristofferC for reviewing.)

**Comment by oschulz** [comment]
I'm also hoping to use this a lot with read-only software containers.

**Pull request merged**